### PR TITLE
build: bump calico to 3.28.0

### PR DESCRIPTION
**Pull request by eg-ayoub**

### CODE REVIEW

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 1]
```diff
@@ -4665,7 +4808,7 @@ spec:
         # It can be deleted if this is a fresh installation, or if you have already
         # upgraded to use calico-ipam.
         - name: upgrade-ipam
-          image: {{ build_image_name('calico-cni') }}
+          image: docker.io/calico/cni:v3.28.0
```
Comment (eg-ayoub) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: change this back
add note

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 2]
```diff
@@ -4793,14 +4936,9 @@ spec:
...
               value: "autodetect"
-            - name: IP_AUTODETECTION_METHOD
-              # NOTE: Use all workload CIDRs
-              value: cidr={{ networks.workload_plane.cidr | join(',') }}
             # Enable IPIP
-            # Note: In MetalK8s we want to use IPIP encapsulation
-            #       only for cross subnet communication.
             - name: CALICO_IPV4POOL_IPIP
-              value: "CrossSubnet"
+              value: "Always"
```
Comment (eg-ayoub) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: change this back

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 3]
```diff
@@ -4828,8 +4966,8 @@ spec:
             # The default IPv4 pool to create on startup if none exists. Pod IPs will be
             # chosen from this range. Changing this value after installation will have
             # no effect. This should fall within `--cluster-cidr`.
-            - name: CALICO_IPV4POOL_CIDR
-              value: "{{ networks.pod }}"
+            # - name: CALICO_IPV4POOL_CIDR
+            #   value: "192.168.0.0/16"
```
Comment (eg-ayoub) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: idem

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 4]
```diff
@@ -4841,10 +4979,6 @@ spec:
               value: "false"
             - name: FELIX_HEALTHENABLED
               value: "true"
-            # Note: We do not want to report about outgoing connections
-            #       in Metalk8s
-            - name: FELIX_USAGEREPORTINGENABLED
-              value: "false"
```
Comment (eg-ayoub) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: put back

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 5]
```diff
@@ -4987,16 +5118,11 @@ spec:
...
-        - key: node-role.kubernetes.io/bootstrap
-          effect: NoSchedule
-        - key: node-role.kubernetes.io/infra
-          effect: NoSchedule
       serviceAccountName: calico-kube-controllers
       priorityClassName: system-cluster-critical
       containers:
         - name: calico-kube-controllers
-          image: {{ build_image_name('calico-kube-controllers') }}
+          image: docker.io/calico/kube-controllers:v3.28.0
```
Comment (eg-ayoub) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: idem

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 6]
```diff
@@ -69,7 +69,6 @@ data:
   #       see: https://github.com/projectcalico/felix/pull/2511#issuecomment-733121759
   # MTU for calico = workload MTU - 20 (for IPinIP header)
   veth_mtu: "{{ networks.workload_plane.mtu - 20 }}"
-
```
Comment (TeddyAndrieux) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: :)

File: salt/metalk8s/kubernetes/cni/calico/deployed.sls [hunk 7]
```diff
@@ -5019,3 +5175,4 @@ spec:
               - /usr/bin/check-status
               - -r
             periodSeconds: 10
+
```
Comment (TeddyAndrieux) [inline salt/metalk8s/kubernetes/cni/calico/deployed.sls]: :)

**Review by TeddyAndrieux — APPROVED** [review]

**Pull request merged**